radix-engine/src/system/system_modules/costing/fee_table.rs

```diff
@@ -564,11 +564,23 @@ impl FeeTable {
         // Based on  `test_crypto_scrypto_verify_secp256k1_ecdsa_costing`
         //   instructions_cnt = 464236 (input is always 32 bytes long)
         //   Lets round:
-        //     444420.94242 -> 446000
-        let instructions_cnt = 446000;
+        //     464236 -> 500000
+        let instructions_cnt = 500000;
         // Convert to cost units
         instructions_cnt / CPU_INSTRUCTIONS_TO_COST_UNIT
     }
+
+    #[inline]
+    pub fn secp256k1_ecdsa_key_recover_cost(&self) -> u32 {
```

Comment:
Why did we base our costing on `test_crypto_scrypto_key_recover_secp256k1_ecdsa`? In that test we create a new ledger simulator, call the simulator, build manifests, among many other things which means that the number of instructions we get from that test is for `secp256k1_key_recovery + ledger_simulator_building + manifest_execution + wasm invocations + etc...`. Wouldn't it be more accurate to calculate the number of instructions that it takes to recover a Secp256k1 natively?

Reply:
In other words, this isn't the "number of instructions to perform Secp256k1 recovery" but rather the "number of instructions to perform Secp256k1 recovery in wasm and execute it via our `LedgerSimulator`"

Reply:
QEMU costing job runs many tests, `test_crypto_scrypto_key_recover_secp256k1_ecdsa` included.

When getting the instruction count we grep QEMU job results for given instruction. eg. `crypto_utils_secp256k1_ecdsa_key_recover`
![image](https://github.com/user-attachments/assets/d95129c6-f613-400d-b13d-c19a063d38d3)
The instruction we want to find must be "traced" (marked with `trace_resources` attribute, see [this](https://github.com/radixdlt/radixdlt-scrypto/pull/1952/files/5a97d1350e181d2359eb514bb146f66799d0dccd#diff-882b9c55afb3d9bed9557b03b333eb2357f3515a16683d19ed91d6f46f1e17f4R749)).

> Wouldn't it be more accurate to calculate the number of instructions that it takes to recover a Secp256k1 natively?

`crypto_utils_secp256k1_ecdsa_key_recover` is traced, which means we do exactly what you suggest :)
So despite there are much more instructions executed for this test, we take into account only those related to `crypto_utils_secp256k1_ecdsa_key_recover`

Reply:
Makes sense to me, thank you for the explanation.